### PR TITLE
Current Homepage Channels Query

### PR DESCRIPTION
**Pull request by ponkey364**
(fyi, this is the first bit of elixr I've ever written, although it's mostly a copy/paste job from other queries)

Done off the back of #719, with the aim of going in and implementing the homepage streams list in the mobile app.

For the most part, I looked at how the other queries work, especially the categories query, so I think it *should* be in line with what's expected here.

I just used a `[Channel]` rather than a connection because I didn't see the point in setting it up for 6 streams

### CODE REVIEW

**Comment by clone1018** [comment]
Thanks for the PR! Looks good so far. 

> I just used a `[Channel]` rather than a connection because I didn't see the point in setting it up for 6 streams

Let's use a connection anyway, as we continue to grow out the homepage we don't want to introduce breaking changes to the API just to add more streams :P. This is what got us into trouble with the first API.

**Comment by ponkey364** [comment]
I think that this attempt should be better, it has a connection now

File: lib/glimesh/api/schema/channel_types.ex [hunk 1]
```diff
@@ -61,6 +61,11 @@ defmodule Glimesh.Api.ChannelTypes do
 
       resolve(&ChannelResolver.all_channels/2)
     end
+
+    @desc "List the channels currently on the homepage"
+    connection field :homepage, node_type: :channel do
```
Comment (clone1018) [inline lib/glimesh/api/schema/channel_types.ex:66]:
```suggestion
    connection field :homepage_channels, node_type: :channel do
```

File: lib/glimesh/api/resolvers/channel_resolver.ex [hunk 2]
```diff
@@ -87,6 +88,15 @@ defmodule Glimesh.Api.ChannelResolver do
 
   def find_channel(_, _), do: {:error, @error_not_found}
 
+  def list_homepage_streams(args, _) do
```
Comment (clone1018) [inline lib/glimesh/api/resolvers/channel_resolver.ex:91]:
```suggestion
  def list_homepage_channels(args, _) do
```

File: lib/glimesh/api/schema/channel_types.ex [hunk 3]
```diff
@@ -61,6 +61,11 @@ defmodule Glimesh.Api.ChannelTypes do
...
+      resolve(&ChannelResolver.list_homepage_streams/2)
```
Comment (clone1018) [inline lib/glimesh/api/schema/channel_types.ex:67]:
```suggestion
      resolve(&ChannelResolver.list_homepage_channels/2)
```

**Review by clone1018 — CHANGES_REQUESTED** [review]
Tested it locally, works great! Ready to merge after a couple of nitpicks. Thanks again!

**Pull request merged**